src/components/TextInput/TextInput.tsx

```diff
@@ -28,27 +31,50 @@ export type TextInputProps = NativeAttributes<'input'> & {
   /** The icon present on the input  */
   icon?: IconProps['type'];
 
+  /** The icon alignment  */
+  iconAlignment?: 'left' | 'right';
+
   /** Additional icon props  */
   iconProps?: TextIconProps;
 };
 
 const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(function TextInput(
-  { label, invalid, required, disabled, id, hidden, name, icon, iconProps = {}, value, ...rest },
+  {
+    label,
+    invalid,
+    required,
+    disabled,
+    id,
+    hidden,
+    name,
+    icon,
+    iconProps = {},
+    variant = 'outline',
+    iconAlignment = 'right',
+    value,
+    ...rest
+  },
   ref
 ) {
   const identifier = id || name || slugify(label);
 
   return (
-    <InputControl invalid={invalid} disabled={disabled} required={required} hidden={hidden}>
-      <Flex align="center" position="relative">
+    <InputControl
+      variant={variant}
+      invalid={invalid}
+      disabled={disabled}
+      required={required}
+      hidden={hidden}
+    >
+      <Flex align="center" position="relative" ml={icon && iconAlignment === 'left' && 30}>
         <InputElement
           as="input"
           type="text"
           id={identifier}
           name={name}
           value={value}
           truncated
-          pr={icon && 10}
+          pr={icon && iconAlignment === 'right' && 10}
```

Comment:
shouldn't we have  a  `pl` (padding left) when `iconAlignment === 'left'`, rather than putting it as margin in the parent component?

Reply:
@3nvi The label is an absolute positioned element thus it's not aligned properly. We should use margins or either wrap the label and the text input with another `<Box />`. Any thoughts on this one?